cloud/aws/aws_s3.cc

```diff
@@ -185,13 +187,12 @@ class AwsS3ClientWrapper {
     }
     return outcome;
   }
-  std::shared_ptr<Aws::Transfer::TransferHandle> DownloadFile(
-      const Aws::String& bucket_name, const Aws::String& object_path,
-      const Aws::String& destination) {
+
+  template <class... Args>
+  std::shared_ptr<Aws::Transfer::TransferHandle> DownloadFile(Args... args) {
```

Comment:
is this change necessary, since the custom stream buf doesn't work with AWS transfer manager?

Reply:
It's not strictly necessary, but it allows the DoGetCloudObject to use a similar code structure for the two paths (specifying the destination using an iostream factory in both cases).